lib/delay-histogram.lua

```diff
@@ -349,15 +349,12 @@ local function calculate_thresholds(histogram_no, print_it, now)
             -- a heuristic in case of relatively low counts (eg. during the long initial build of the histogram)
             -- find the lowest bucket with a count > 1
             if histogram[result] == 1 then
-                local r = min_allowed_threshold
-                for i = min_allowed_threshold + 1, max_allowed_threshold do
-                    if histogram[result] > 1 then
-                        r = i
+                for i = result,min_allowed_threshold,-1 do
```

Comment:
This change does not achieve the required heuristic but does point out an optimisation:
`for i = min_allowed_threshold + 1, result -1 do